From 5d0a66692e256dd9f72bbc24bfffba196547cee3 Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Wed, 28 Jul 2021 13:52:11 +0530
Subject: [PATCH 3/4] octeontx2-pf: Fix default DWRR quantum configuration

commit 1317bf81a5f7f3027e2d0d0f76a34babafd5079d from
git@git.assembla.com:cavium/WindRiver.linux.git

By default all the transmit queues are configured to transmit
pkts via same scheduler hierarchy ie SMQ->TL4->TL3->TL2->TL1.
So HW does a DWRR between all transmit queues. The DWRR quantum
being configured currently is too high and could lead to unfair
scheduling between transmit queues. Fixed this by setting quantum
to 16K which is close to 2x of max frame size supported by
OcteonTx2 HW.

Change-Id: If866d7ce25e406f480ef32baf403817ea5c1c02d
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/57830
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/common.h | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/common.h b/drivers/net/ethernet/marvell/octeontx2/af/common.h
index 770fa4bf00d7..737ed0c9dcc0 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/common.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/common.h
@@ -146,10 +146,7 @@ enum nix_scheduler {
 #define TXSCH_RR_QTM_MAX		((1 << 24) - 1)
 #define TXSCH_TL1_DFLT_RR_QTM		TXSCH_RR_QTM_MAX
 #define TXSCH_TL1_DFLT_RR_PRIO		(0x1ull)
-#define MAX_SCHED_WEIGHT		0xFF
-#define DFLT_RR_WEIGHT			71
-#define DFLT_RR_QTM	((DFLT_RR_WEIGHT * TXSCH_RR_QTM_MAX) \
-			 / MAX_SCHED_WEIGHT)
+#define DFLT_RR_QTM			16384 /* ~2x of MAX MTU on OcteonTx2 */
 
 /* Min/Max packet sizes, excluding FCS */
 #define	NIC_HW_MIN_FRS			40
-- 
2.31.1

